### PR TITLE
symbolic_shapes.py extra variants

### LINKED ISSUE #119969: Also attempt to match weaker conditions when replacing deferred runtime asserts
### 🐛 Describe the bug

Internal xref: https://fb.workplace.com/groups/6829516587176185/posts/6853529854774858/?comment_id=6856660057795171&notif_id=1707934443990760&notif_t=work_group_comment_mention

People complaining that `torch._check(u1 < u2)` doesn't work to fix a GuardOnUnbackedSymNode error, but the weaker `torch._check(u1 <= u2)`. This particular problem is easy to resolve in symbolic_shapes.py

```
        # Apply known runtime asserts
        for s in symbols:
            # Unbacked symints only
            if s in self.var_to_val:
                continue
            subst = {}
            for ra in self.deferred_runtime_asserts.get(s, ()):
                if compute_hint:
                    e = canonicalize_bool_expr(ra.expr.xreplace(self.var_to_val))
                else:
                    e = ra.expr
                # e is already canonical
                subst[e] = sympy.true
                subst[canonicalize_bool_expr(sympy.Not(e))] = sympy.false
                if isinstance(e, sympy.Eq):
                    subst[sympy.Le(e.lhs, e.rhs)] = sympy.true
                    subst[sympy.Le(-e.lhs, -e.rhs)] = sympy.true
                    subst[sympy.Lt(e.lhs, e.rhs)] = sympy.false
                    subst[sympy.Lt(-e.lhs, -e.rhs)] = sympy.false
```

This is the current replacement. We should generate more variants when we are given sympy.Lt, similar to what we do with Eq.

cc @msaroufim @bdhirsh @anijain2305 @zou3519 @lezcano @avikchaudhuri 

### Versions

main

### PR DESCRIPTION
**Pull request by mingxzhao**
Fixes #119969

Adds extra variants for sympy.Lt

### CODE REVIEW

**Comment by ezyang** [comment]
A test would be nice

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -3333,6 +3333,13 @@ def _maybe_evaluate_static(
                     subst[sympy.Le(-e.lhs, -e.rhs)] = sympy.true
                     subst[sympy.Lt(e.lhs, e.rhs)] = sympy.false
                     subst[sympy.Lt(-e.lhs, -e.rhs)] = sympy.false
+                elif isinstance(e, sympy.Lt):
+                    subst[sympy.Le(e.lhs, e.rhs)] = sympy.true
+                    subst[sympy.Le(-e.lhs, -e.rhs)] = sympy.true
```
Comment (ezyang) [inline torch/fx/experimental/symbolic_shapes.py:3338]: If `a < b`, then it is NOT true that `-a < -b`, indeed, you need to flip the inequality

**Review by ezyang — CHANGES_REQUESTED** [review]
seems wrong

File: torch/fx/experimental/symbolic_shapes.py [hunk 2]
```diff
@@ -3333,6 +3333,12 @@ def _maybe_evaluate_static(
                     subst[sympy.Le(-e.lhs, -e.rhs)] = sympy.true
                     subst[sympy.Lt(e.lhs, e.rhs)] = sympy.false
                     subst[sympy.Lt(-e.lhs, -e.rhs)] = sympy.false
+                elif isinstance(e, sympy.Lt):
+                    subst[sympy.Lt(e.lhs, e.rhs)] = sympy.true
+                    subst[sympy.Le(e.lhs, e.rhs)] = sympy.true
+                    subst[sympy.Gt(e.lhs, e.rhs)] = sympy.false
```
Comment (lezcano) [inline torch/fx/experimental/symbolic_shapes.py:3339]: The canonicalised expressions do not have `Gt`, so this is unnecessary

File: torch/fx/experimental/symbolic_shapes.py [hunk 3]
```diff
@@ -3333,6 +3333,12 @@ def _maybe_evaluate_static(
                     subst[sympy.Le(-e.lhs, -e.rhs)] = sympy.true
                     subst[sympy.Lt(e.lhs, e.rhs)] = sympy.false
                     subst[sympy.Lt(-e.lhs, -e.rhs)] = sympy.false
+                elif isinstance(e, sympy.Lt):
+                    subst[sympy.Lt(e.lhs, e.rhs)] = sympy.true
```
Comment (lezcano) [inline torch/fx/experimental/symbolic_shapes.py:3337]: This is already added above

File: torch/fx/experimental/symbolic_shapes.py [hunk 4]
```diff
@@ -3333,6 +3333,12 @@ def _maybe_evaluate_static(
...
+                    subst[sympy.Le(e.lhs, e.rhs)] = sympy.true
+                    subst[sympy.Gt(e.lhs, e.rhs)] = sympy.false
+                    subst[sympy.Ge(e.lhs, e.rhs)] = sympy.false
```
Comment (lezcano) [inline torch/fx/experimental/symbolic_shapes.py:3340]: Same as Gt

File: torch/fx/experimental/symbolic_shapes.py [hunk 5]
```diff
@@ -3333,6 +3333,12 @@ def _maybe_evaluate_static(
...
+                    subst[sympy.Ne(e.lhs, e.rhs)] = sympy.true
```
Comment (lezcano) [inline torch/fx/experimental/symbolic_shapes.py:3341]: We should probably canonicalise `Ne` into `Or(Lt(a,b), Lt(b, a))`, but I guess this is fine for now.

**Review by lezcano — COMMENTED** [review]
There are a number of equations still missing

**Comment by lezcano** [comment]
It's still missing a test.

**Comment by mingxzhao** [comment]
Yes apologies, working on test soon. Just wanted to resolve the comment, not quite ready for review yet.

**Pull request closed without merging**